plugins/techdocs-addons-test-utils/src/test-utils.tsx

```diff
@@ -0,0 +1,240 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, Fragment } from 'react';
+
+// Shadow DOM support for the simple and complete DOM testing utilities
+// https://github.com/testing-library/dom-testing-library/issues/742#issuecomment-674987855
+import { screen } from 'testing-library__dom';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Route } from 'react-router-dom';
+import { act, render } from '@testing-library/react';
+
+import {
+  wrapInTestApp,
+  TestApiProvider,
+  TestApiProviderProps,
+} from '@backstage/test-utils';
+import { FlatRoutes } from '@backstage/core-app-api';
+
+import {
+  TechDocsAddons,
+  techdocsApiRef,
+  TechDocsEntityMetadata,
+  TechDocsMetadata,
+  techdocsStorageApiRef,
+} from '@backstage/plugin-techdocs-react';
+import { TechDocsReaderPage, techdocsPlugin } from '@backstage/plugin-techdocs';
+import { catalogPlugin } from '@backstage/plugin-catalog';
+import { searchApiRef } from '@backstage/plugin-search-react';
+
+/**
+ * @public
+ */
+export type RecursivePartial<T> = {
+  [P in keyof T]?: RecursivePartial<T[P]>;
+};
+
+const techdocsApi = {
+  getTechDocsMetadata: jest.fn(),
+  getEntityMetadata: jest.fn(),
+};
+
+const techdocsStorageApi = {
+  getApiOrigin: jest.fn(),
+  getEntityDocs: jest.fn(),
+  syncEntityDocs: jest.fn(),
+};
+
+const searchApi = {
+  query: jest.fn().mockResolvedValue({ results: [] }),
+};
+
+/**
+ * @public
+ */
+export type Apis = TestApiProviderProps<any>['apis'];
+
+type TechDocsAddonBuilderOptions = {
+  dom: ReactElement;
+  entity: RecursivePartial<TechDocsEntityMetadata>;
+  metadata: RecursivePartial<TechDocsMetadata>;
+  componentId: string;
+  apis: Apis;
+  path: string;
+};
+
+const defaultOptions: TechDocsAddonBuilderOptions = {
+  dom: <></>,
+  entity: {},
+  metadata: {},
+  componentId: 'docs',
+  apis: [],
+  path: '',
+};
+
+const defaultMetadata = {
+  site_name: 'Tech Docs',
+  site_description: 'Tech Docs',
+};
+
+const defaultEntity = {
+  kind: 'Component',
+  metadata: { namespace: 'default', name: 'docs' },
+};
+
+const defaultDom = (
+  <html lang="en">
+    <head />
+    <body>
+      <div data-md-component="container">
+        <div data-md-component="navigation" />
+        <div data-md-component="toc" />
+        <div data-md-component="main" />
+      </div>
+    </body>
+  </html>
+);
+
+/**
+ * @public
+ */
+
+export class TechDocsAddonBuilder {
+  private options: TechDocsAddonBuilderOptions = defaultOptions;
+  private addons: ReactElement[];
+
+  static buildAddonsInTechDocs(addons: ReactElement[]) {
+    return new TechDocsAddonBuilder(addons);
+  }
+
+  constructor(addons: ReactElement[]) {
+    this.addons = addons;
+  }
+
+  withApis(apis: Apis) {
```

Comment:
Perhaps skip this method? Feels like it already is simple enough to just wrap with a `TestApiProvider`, and that method lets you create and use a registry too if you want. If you feel strongly about keeping it I think it's best to make the type here be `TestApiProviderProps<T>['apis']`

Reply:
We decided to keep this because the `buildAddonsInTechDocs` method is intended to take an array of addons, and wrapping just _them_ in their own `<TestApiProvider>` is a bit odd.  Types will be fixed up though.  (Fixes coming)